pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentStreamObserver.java

```diff
@@ -64,6 +65,12 @@ public MailboxContentStreamObserver(GrpcMailboxService mailboxService,
     _isEnabledFeedback = isEnabledFeedback;
   }
 
+  /**
+   * This method may return null. It can happen if there's a large enough gap between the time the receiver received
+   * the last block and the StreamObserver was detected as complete. In that case, the MailboxReceiveOperator would
+   * try to call this method again but since there's no new content to be received, we'll exit the loop. Returning
+   * null here means that MailboxReceiveOperator won't consider this as an error.
+   */
```

Comment:
@walterddr : Highlighting this since we wanted to discuss about when poll here would return null. I think there's only 1 case which I have highlighted above. Lmk your thoughts.

Also an interesting thing I saw was that we are not processing `MAILBOX_METADATA_END_OF_STREAM_KEY` on the receiver side, though the sender does set it.

Reply:
1. yes this is the correct behavior.
2. we don't rely on this b/c we rely on `onCompleted()`